Ensure that LIBV4L_HAVE_POSIX_IOCTL define is used when linking with other packages.

diff -rup meson.build.orig meson.build
--- a/meson.build.orig
+++ b/meson.build
@@ -383,12 +383,14 @@ if have_qt_opengl
     conf.set('HAVE_QTGL', 1)
 endif
 
+pkg_arguments = []
 ioctl_posix_test = '''
 #include <sys/ioctl.h>
 int ioctl (int, int, ...);
 '''
 if cc.compiles(ioctl_posix_test)
     conf.set('LIBV4L_HAVE_POSIX_IOCTL', 1)
+    pkg_arguments += '-DLIBV4L_HAVE_POSIX_IOCTL=1'
 endif
 
 c_arguments = []
diff -rup lib/libv4l2/meson.build.orig lib/libv4l2/meson.build
--- a/lib/libv4l2/meson.build.orig
+++ b/lib/libv4l2/meson.build
@@ -48,7 +48,8 @@ pkg.generate(
     name : 'libv4l2',
     version : meson.project_version(),
     requires_private : 'libv4lconvert',
-    description : 'v4l2 device access library')
+    description : 'v4l2 device access library',
+    extra_cflags : pkg_arguments)
 
 if not get_option('v4l-wrappers')
     subdir_done()

